packages/devcmd/src/process/ProcessExecutor.ts

```diff
@@ -183,16 +228,25 @@ function childProcessCompletion(childProcess: ChildProcess): Promise<number | nu
   });
 }
 
-type Result = { ok: true } | { err: unknown };
+type Ok = { ok: true };
+type Err = { err: unknown };
+
+type Result = Ok | Err;
 
-function isErr(r: Result): r is { err: unknown } {
+function isErr(r: Result): r is Err {
   return !isOk(r);
 }
 
-function isOk(r: Result): r is { ok: true } {
+function isOk(r: Result): r is Ok {
   return "ok" in r && r.ok;
 }
 
+/**
+ * Executes an awaitable function and wraps the result of the Promise
```

Comment:
nit-pick: in JS/TS, every value is awaitable (it just immediately yields the value itself, e.g. `await 2` works and evaluates to 2). maybe "a promise-returning function" is more apt ;)

Reply:
Damn it TypeScript, why cant you be easy on me one single time ;-)